backend/app/routes/web/recommendation.py

```diff
@@ -76,47 +72,45 @@ def __str__(self):
         return self.__repr__()
 
 
-def _get_panel(panel_id: RecommendationPanelId) -> list[str]:
+def _get_panel(panel_id: RecommendationPanelId) -> str:
     # IMPORTANT!!! recommender does not support services
     panel_id_options = ["publications", "datasets", "software", "trainings"]
 
     match panel_id:
         case "all":
-            return [random.choice(panel_id_options)]
+            return random.choice(panel_id_options)
         case "publication":
-            return ["publications"]
+            return "publications"
         case "dataset":
-            return ["datasets"]
+            return "datasets"
         case "software":
-            return ["software"]
+            return "software"
         case "training":
-            return ["trainings"]
+            return "trainings"
         case "service":
-            return []
-    raise ValueError(f"{panel_id} is not valid {RecommendationPanelId}")
+            return "service"
+        case "data-source":
+            return "data-source"
+    raise RecommenderError(f"{panel_id} is not valid {RecommendationPanelId}")
 
 
 async def _get_recommended_uuids(
     client: AsyncClient, session: SessionData | None, panel_id: RecommendationPanelId
 ):
     try:
-        page_id = "/search/" + panel_id
-        panels = _get_panel(panel_id)
-
-        if not panels:
-            return []
-
         request_body = {
-            "user_id": session.aai_id if session else None,
             "unique_id": session.session_uuid if session else str(uuid.uuid4()),
             "timestamp": datetime.datetime.utcnow().isoformat()[:-3] + "Z",
             "visit_id": str(uuid.uuid4()),
-            "page_id": page_id,
-            "panel_id": panels,
+            "page_id": "/search/" + panel_id,
+            "panel_id": _get_panel(panel_id),
             "candidates": [],
             "search_data": {},
         }
 
+        if session:
+            request_body["aai_id"] = session.aai_id
```

Comment:
In user actions as `aai_uid` `session.aai_state` is being sent. Is this correct? @Michal-Kolomanski ?

Reply:
Yes, it is. But what is the exact difference between `session.aai_state `and `session.aai_id`?

Reply:
`state` zwracany jest przez query param z checkina, `aai_id` jest brany z `aai_id = aai_response["userinfo"]["sub"]`

```
async def auth_checkin(code: str, state: str):
    if not state:
        return RedirectResponse(status_code=400, url=UI_BASE_URL)

    try:
        aai_response = rp_handler.finalize(OIDC_ISSUER, dict(code=code, state=state))

        session_id = uuid4()
        username = aai_response["userinfo"]["name"]
        aai_id = aai_response["userinfo"]["sub"]

        session_data = SessionData(
            username=username,
            aai_state=state,
            aai_id=aai_id,
            session_uuid=str(uuid.uuid4()),
        )
```

Reply:
We should pass here the same AAI_UID which is associated with the user in the MP's DB. This ID is used to uniquely identify the user across all services. That being said, that has remained me that there is a bug causing duplication of users with the same AAI_UID. We should fix it in the new sprint.